hooks/gen_docs/__init__.py

```diff
@@ -1,30 +1,23 @@
 """Documentation generation."""
 
-from collections.abc import Generator
+from collections.abc import Iterator
 from enum import Enum
-from typing import Any, Generic, TypeVar
 
-_T = TypeVar("_T")
 
-
-class SuperEnum(Generic[_T], Enum):
-    """Adds constructors that return all items in a ``Generator``.
-
-    Introduces constructors that return a ``Generator`` object
-    either containing all items, only their names or their values.
-    """
+class IterableStrEnum(str, Enum):
+    """Temporary replacement for StrEnum while we suppport python3.10"""
```

Comment:
it's not just that, no?
that would be a fitting description for a simple polyfill like `class StrEnum(str, Enum): ...`
but here the focus is more on the extra functionality that makes it similar to a dict, wdyt?

Reply:
Hmm you are right. I had a (wrong) memory that `StrEnum` already had that functionality somehow built in.